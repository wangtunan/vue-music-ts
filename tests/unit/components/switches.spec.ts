import { shallowMount, Wrapper } from '@vue/test-utils'
import Switches from '@/components/switches/index.vue'

describe('switches.vue', () => {
  let wrapper: Wrapper<Vue>
  beforeEach(() => {
    wrapper = shallowMount(Switches, {
      propsData: {
        active: 1,
        switches: ['我喜欢的', '最近听的']
      }
    })
  })
  it('match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
  it('render switches array success', () => {
    const switchItems = wrapper.findAll('.switch-item')
    expect(switchItems.length).toBe(2)
    expect(switchItems.at(1).text()).toBe('最近听的')
  })
  it('bind active class', () => {
    const switchItems = wrapper.findAll('.switch-item')
    expect(switchItems.at(1).classes('active')).toBe(true)
  })
})