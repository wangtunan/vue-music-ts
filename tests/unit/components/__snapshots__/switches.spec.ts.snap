// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`switches.vue match snapshot 1`] = `
<ul class="m-switches">
  <li class="switch-item">我喜欢的</li>
  <li class="switch-item active">最近听的</li>
</ul>
`;
